src/os_posix/os_mtx_rw.c

```diff
@@ -79,9 +80,11 @@ __wt_try_readlock(WT_SESSION_IMPL *session, WT_RWLOCK *rwlock)
 	l = &rwlock->rwlock;
 	pad = l->s.pad;
 	users = l->s.users;
+	nusers = (uint16_t)users + 1;
 	writers = l->s.writers;
 	old = (pad << 48) + (users << 32) + (users << 16) + writers;
-	new = (pad << 48) + ((users + 1) << 32) + ((users + 1) << 16) + writers;
+	/* Intentionally overflow (users + 1) into pad to handle 64K lockers. */
+	new = (pad << 48) + ((users + 1) << 32) + (nusers << 16) + writers;
```

Comment:
@keithbostic, perhaps it could be done with casting or bit masking:
```
new = (pad << 48) + ((users + 1) << 32) + (((users + 1) & 0xffff) << 16) + writers;
```